pkg/controller/common/name/name.go

```diff
@@ -7,20 +7,42 @@ package name
 import (
 	"fmt"
 	"strings"
+	"unicode/utf8"
 
 	"github.com/elastic/cloud-on-k8s/pkg/utils/stringsutil"
+	"github.com/hashicorp/go-multierror"
+	"k8s.io/apimachinery/pkg/util/validation"
 	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
 )
 
-var (
-	log = logf.Log.WithName("name")
-)
-
 const (
-	// MaxNameLength is the maximum length of a resource name.
-	MaxNameLength = 63
+	MaxResourceNameLength = 36
```

Comment:
It would be good to add a comment here to explain why this magic `36`, maybe something like: arbitrary length, maps that of a standard UUID and gives us some room to add a suffix without reaching the maximum length of a label value.